Bottom: 11ede0a01c80719ab9de48a1ca2aa6e9ba2890df
Top:    ee0f8cc03d6be387bbec3ca0b3297ea6b0f51bd7
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-02-13 17:14:44 +0000

Refactor polygon copy operations


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 14d6c78..c1861d8 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2990,13 +2990,13 @@ BOOLp
 poly_Copy0 (POLYAREA ** dst, const POLYAREA * src)
 {
   *dst = NULL;
-  if (src != NULL)
-    *dst = (POLYAREA *)calloc (1, sizeof (POLYAREA));
-  if (*dst == NULL)
+  if (src == NULL)
+    return TRUE;
+
+  if ((*dst = poly_Create ()) == NULL || !poly_Copy1 (*dst, src))
     return FALSE;
-  (*dst)->contour_tree = r_create_tree (NULL, 0, 0);
 
-  return poly_Copy1 (*dst, src);
+  return TRUE;
 }
 
 void
